src/python/pants/util/rwbuf.py

```diff
@@ -34,8 +35,10 @@ def read_from(self, pos, size=-1):
       return self._io.read() if size == -1 else self._io.read(size)
 
   def write(self, s):
+    if not isinstance(s, bytes):
```

Comment:
@benjyw follow up from prior discussion. Good point about prior version changing semantics.

This does change semantics that now `ValueError` will be raised whereas earlier we would try to coerce into bytes. If it's safe to make this change, I agree with you this is better

Reply:
True, but that coercion might have failed, so it's best to at least fail predictably and eagerly.